### PR TITLE
Feature/jel 329 modularize login system

### PR DESCRIPTION
**Pull request by OBCENEIKON**
This adds jellyfish-a-fied omniauth plugin compatibility.

### CODE REVIEW

File: app/models/staff.rb [hunk 1]
```diff
@@ -63,4 +64,21 @@ class Staff < ActiveRecord::Base
   def gravatar
     '3fc88b95c85e43f157cb1ffd0e37e832'
```
Comment (chriskacerguis) [inline app/models/staff.rb]: I know that you didn't do this, but what is this for?

File: spec/factories/authentications.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#  uid        :string
+#
+# Indexes
+#
+#  index_authentications_on_provider_and_uid  (provider,uid)
+#  index_authentications_on_staff_id          (staff_id)
+#
+
+FactoryGirl.define do
+  factory :authentications do
```
Comment (chriskacerguis) [inline spec/factories/authentications.rb]: Do we need this FB stuff?

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -14,7 +14,15 @@ class SessionsController < Devise::SessionsController
   def create
     respond_to do |format|
       format.html do
-        super
+        if request.env['omniauth.auth']
+          auth_hash = request.env['omniauth.auth']
+
+          staff = Staff.find_by_auth(auth_hash)
+
+          sign_in_and_redirect(resource_name, staff) unless staff.nil?
```
Comment (calebthompson) [inline app/controllers/sessions_controller.rb]: double negative here... can you do `if staff.present?` instead?

File: app/models/staff.rb (hunk 1 above)
Comment (OBCENEIKON) [inline app/models/staff.rb]: It looks like it's a method that just returns some static data. I found references to it in multiple places.
It looks like it's a method that just returns some static data. I found references to it in multiple places.
[config/routes.rb:34](https://github.com/projectjellyfish/api/blob/2429a1c45c05db1b729683812a158b590c13f969/config/routes.rb#L34)
[app/controllers/staff_controller.rb:17](https://github.com/projectjellyfish/api/blob/2429a1c45c05db1b729683812a158b590c13f969/app/controllers/staff_controller.rb#L17)
[doc/apipie_examples.json:3612, 3627, 3642](https://github.com/projectjellyfish/api/blob/2429a1c45c05db1b729683812a158b590c13f969/app/controllers/staff_controller.rb#L3612-3642)

I don't see a column for it in the staff table, you want me to remove it?

Comment (chriskacerguis) [inline app/models/staff.rb]: If we don't need it, then yeah let's clean it out (along with that other stuff)

**Comment by OBCENEIKON** [comment]
Tested it, it does indeed work :)

**Pull request merged**